components/mdns/CMakeLists.txt

```diff
@@ -10,15 +10,22 @@ else()
     set(MDNS_CONSOLE "")
 endif()
 
+if (CONFIG_MDNS_MEMORY_ALLOC_CUSTOM_IMPL)
+    # User has to provide their own implementation of memory allocation functions
+    set(MDNS_MEMORY "")
```

Comment:
Nits: Do we need an additional configuration option to specify the directory path for the user's implementation file? Users might face difficulties adding their files to the mDNS CMakeLists, as modifying the downloaded component is not allowed.

Reply:
> Do we need an additional configuration option to specify the directory path for the user's implementation file?

Not really, the idea here is to provide an implementation in an external component (like `main` or some other user side component) and set a link dependency to mdns (which can be done through project makefile)

But after thinking about it again, maybe an easier option would be to revert this config and just mark the mdns memory functions as `__weak`, WYT ?

Reply:
Ok, agree with marking the mdns memory functions as `__weak` if macro `CONFIG_MDNS_MEMORY_ALLOC_CUSTOM_IMPL` is enabled.

Reply:
ok, made the memory functions optionally `weak`.
also allowing to rewrite the task allocators, which is useful for setting some static reserved space for the task stack.